### ISSUE TITLE
[osx] Crash on empty string as font specification

### ISSUE THREAD

**Issue by dschreij** [smathot/OpenSesame#397]
On OSX, it is possible that QProgEdit is initialized with an empty string as the font to use. This results in an exception specifying that an unexpected QPyNullVariant has been received instead of string specifying a font. Possibly this is caused by an out-of-date configuration file which can be reset with opensesame --start-clean?

```
22/02/16 21:27:05,023 Console[4862]:  Marker - 22 Feb 2016 21:27:05
22/02/16 21:27:09,292 Console[4862]:  Marker - 22 Feb 2016 21:27:09
22/02/16 21:27:14,932 Console[4862]:  Marker - 22 Feb 2016 21:27:14
22/02/16 21:27:18,646 OpenSesame[4876]: ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to /var/folders/sp/0zg5nhg15y1d6kkr3m6n_z740000gn/T/nl.cogsci.osdoc.savedState
22/02/16 21:27:18,646 OpenSesame[4876]: 2016-02-22 21:27:18.645 OpenSesame[4876:1406840] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to /var/folders/sp/0zg5nhg15y1d6kkr3m6n_z740000gn/T/nl.cogsci.osdoc.savedState
22/02/16 21:27:18,654 appleeventsd[54]: SecTaskLoadEntitlements failed error=22
22/02/16 21:27:18,658 launchservicesd[79]: SecTaskLoadEntitlements failed error=22
22/02/16 21:27:18,661 launchservicesd[79]: SecTaskLoadEntitlements failed error=22
22/02/16 21:27:21,026 OpenSesame[4876]: WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.3 instead of 10.11.3. This is not a bug in Gestalt -- it is a documented limitation. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.
Call location:
22/02/16 21:27:21,027 OpenSesame[4876]: 0   CarbonCore                          0x00007fff90b11c9b ___Gestalt_SystemVersion_block_invoke + 113
22/02/16 21:27:21,027 OpenSesame[4876]: 1   libdispatch.dylib                   0x00007fff993ac33f _dispatch_client_callout + 8
22/02/16 21:27:21,027 OpenSesame[4876]: 2   libdispatch.dylib                   0x00007fff993ac237 dispatch_once_f + 67
22/02/16 21:27:21,027 OpenSesame[4876]: 3   CarbonCore                          0x00007fff90a9db47 _Gestalt_SystemVersion + 987
22/02/16 21:27:21,027 OpenSesame[4876]: 4   CarbonCore                          0x00007fff90a9cddb Gestalt + 139
22/02/16 21:27:21,027 OpenSesame[4876]: 5   libQtCore.4.8.7.dylib               0x0000000105b7d11c _ZN16QSettingsPrivate6createEN9QSettings6FormatENS0_5ScopeERK7QStringS5_ + 70
22/02/16 21:27:21,027 OpenSesame[4876]: 6   libQtCore.4.8.7.dylib               0x0000000105b5a2a6 _ZN9QSettingsC1ERK7QStringS2_P7QObject + 28
22/02/16 21:27:22,525 launchservicesd[79]: SecTaskLoadEntitlements failed error=22
22/02/16 21:27:24,660 OpenSesame[4876]: Traceback (most recent call last):
22/02/16 21:27:24,660 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/__boot__.py", line 351, in <module>
22/02/16 21:27:24,660 OpenSesame[4876]:     _run()
22/02/16 21:27:24,660 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/__boot__.py", line 336, in _run
22/02/16 21:27:24,660 OpenSesame[4876]:     exec(compile(source, path, 'exec'), globals(), globals())
22/02/16 21:27:24,660 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/opensesame.py", line 97, in <module>
22/02/16 21:27:24,660 OpenSesame[4876]:     opensesame.resume_init()
22/02/16 21:27:24,660 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/qtopensesame.py", line 208, in resume_init
22/02/16 21:27:24,661 OpenSesame[4876]:     fd.read())
22/02/16 21:27:24,661 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/experiment.py", line 79, in __init__
22/02/16 21:27:24,661 OpenSesame[4876]:     fullscreen=None)
22/02/16 21:27:24,661 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libopensesame/experiment.py", line 168, in __init__
22/02/16 21:27:24,661 OpenSesame[4876]:     item.item.__init__(self, name, self, string)
22/02/16 21:27:24,661 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libopensesame/item.py", line 68, in __init__
22/02/16 21:27:24,662 OpenSesame[4876]:     self.from_string(string)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libopensesame/experiment.py", line 325, in from_string
22/02/16 21:27:24,662 OpenSesame[4876]:     self.items.new(item_type, item_name, def_str)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/misc/qtitem_store.py", line 83, in new
22/02/16 21:27:24,662 OpenSesame[4876]:     item = super(qtitem_store, self).new(_type, name=name, script=script)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libopensesame/item_store.py", line 178, in new
22/02/16 21:27:24,662 OpenSesame[4876]:     item = item_class(name, self.experiment, script)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/sequence.py", line 52, in __init__
22/02/16 21:27:24,662 OpenSesame[4876]:     qtplugin.__init__(self)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/qtplugin.py", line 69, in __init__
22/02/16 21:27:24,662 OpenSesame[4876]:     qtitem.qtitem.__init__(self)
22/02/16 21:27:24,662 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/qtitem.py", line 52, in __init__
22/02/16 21:27:24,662 OpenSesame[4876]:     self.init_edit_widget()
22/02/16 21:27:24,663 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/sequence.py", line 59, in init_edit_widget
22/02/16 21:27:24,663 OpenSesame[4876]:     super(sequence, self).init_edit_widget(False)
22/02/16 21:27:24,663 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/libqtopensesame/items/qtitem.py", line 246, in init_edit_widget
22/02/16 21:27:24,663 OpenSesame[4876]:     self._script_widget.addTab(u'Script').setLang(u'OpenSesame')
22/02/16 21:27:24,663 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qtabmanager.py", line 124, in addTab
22/02/16 21:27:24,663 OpenSesame[4876]:     progEdit = QProgEdit(self, title=title)
22/02/16 21:27:24,667 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qprogedit.py", line 65, in __init__
22/02/16 21:27:24,667 OpenSesame[4876]:     self.editor = QEditor(self, **editorParams)
22/02/16 21:27:24,668 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qeditor.py", line 59, in __init__
22/02/16 21:27:24,668 OpenSesame[4876]:     self.setLang()
22/02/16 21:27:24,668 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qeditor.py", line 452, in setLang
22/02/16 21:27:24,668 OpenSesame[4876]:     colorScheme=self.cfg.qProgEditColorScheme)
22/02/16 21:27:24,668 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qlexer.py", line 212, in QLexer
22/02/16 21:27:24,669 OpenSesame[4876]:     lexer.setTheme(editor, colorScheme)
22/02/16 21:27:24,669 OpenSesame[4876]:   File "/Applications/OpenSesame.app/Contents/Resources/lib/python2.7/QProgEdit/_qlexer.py", line 59, in setTheme
22/02/16 21:27:24,669 OpenSesame[4876]:     self.editor.cfg.qProgEditFontSize)
22/02/16 21:27:24,669 OpenSesame[4876]: TypeError: arguments did not match any overloaded call:
22/02/16 21:27:24,669 OpenSesame[4876]:   QFont(): too many arguments
22/02/16 21:27:24,669 OpenSesame[4876]:   QFont(QString, int pointSize=-1, int weight=-1, bool italic=False): argument 1 has unexpected type 'QPyNullVariant'
22/02/16 21:27:24,669 OpenSesame[4876]:   QFont(QFont, QPaintDevice): argument 1 has unexpected type 'QPyNullVariant'
22/02/16 21:27:24,671 OpenSesame[4876]:   QFont(QFont): argument 1 has unexpected type 'QPyNullVariant'
22/02/16 21:27:24,671 OpenSesame[4876]:   QFont(QVariant): too many arguments
22/02/16 21:27:24,685 OpenSesame[4876]: OpenSesame Error
22/02/16 21:27:24,685 OpenSesame[4876]: 2016-02-22 21:27:24.684 OpenSesame[4876:1406840] OpenSesame Error
22/02/16 21:27:24,825 launchservicesd[79]: SecTaskLoadEntitlements failed error=22
22/02/16 21:27:24,828 launchservicesd[79]: SecTaskLoadEntitlements failed error=22
```

**Comment by JdenHartog** [comment]
Confirmed: running *OpenSesame.app/Contents/MacOS/OpenSesame --start-clean* solves this on my MacBookPro5,3 running OSX10.11.3(El Capitan) and opensesame_3.0.5-osx-3.dmg

**Comment by smathot** [comment]
Should be fixed as of 3.0.6a2.
